test/lib/xml_checks_test.rb

```diff
@@ -57,56 +58,67 @@
 
     context "when the profile is a valid one" do
       it "returns true" do
-        expect(described_class.check(xml, schema, "title")).to eq(true)
+        expect(subject.check(xml, schema, "title")).to eq(true)
       end
     end
 
     context "when the profile is an invalid one" do
+      let(:new_errors) { false }
       let(:valid) { false }
 
       context "and it is the first time that the errors have been reported" do
-        let(:errors_stored?) { false }
+        let(:new_errors) { true }
         let(:popup_selection) { :cancel }
+
         before do
           allow(Yast2::Popup).to receive(:show).and_return(popup_selection)
+          allow(subject).to receive(:new_errors?).and_return(new_errors)
+          allow(subject).to receive(:store_errors)
```

Comment:
to be honest I do not like much such mocking. Is possible to clean instance and instead mock reading of file? So really outer bounds?

Reply:
Yep, I was already thinking on also avoid the mock and check real store and load of the erorrs. Will modify it.